dashboard/app/models/user.rb

```diff
@@ -555,6 +555,13 @@ def memoized_teachers
 
   before_save :remove_cleartext_emails, if: -> {student? && migrated? && user_type_changed?}
 
+  validate :no_family_name_for_teachers, if: -> {DCDO.get('family-name-features', false)}
+  def no_family_name_for_teachers
+    if family_name && (teacher? || sections_as_pl_participant.any?)
```

Comment:
nit: not sure if you need `sections_as_pl_participant` here or not. Conceptually, students shouldn't be a pl participant but I don't know if we explicitly prevent that

Reply:
hmm, in the unit tests I'm creating student users and assigning them to PL sections, so I think there's not validation for that part either? i did see when I tried to add a student to a PL section from the UI there was an error message about permissions, but I'm not sure where it was coming from.